libsolidity/formal/VariableUsage.h

```diff
@@ -17,33 +17,35 @@
 
 #pragma once
 
-#include <map>
-#include <set>
+#include <libsolidity/ast/ASTVisitor.h>
+
 #include <vector>
+#include <set>
 
 namespace dev
 {
 namespace solidity
 {
 
-class ASTNode;
-class VariableDeclaration;
-
 /**
- * This class collects information about which local variables of value type
- * are modified in which parts of the AST.
+ * This class computes information about which variables are modified in a certain subtree.
  */
-class VariableUsage
+class VariableUsage: private ASTConstVisitor
 {
 public:
-	explicit VariableUsage(ASTNode const& _node);
-
-	std::vector<VariableDeclaration const*> touchedVariables(ASTNode const& _node) const;
+	/// @param _outerFunction the function where the current control-flow is in.
```

Comment:
```suggestion
	/// @param _outerFunction the function which contains the current control-flow.
```
:-D - but nevermind that :-D.

So do I understand this correctly in that it would have worked and terminated before, but now it possibly does sooner, since roughly speaking I pass along half of a possibly round-trip that I'd otherwise have had to walk again? If so, I still think this is fine and better now, yeah!

Reply:
Fixed